### PR TITLE
add back job parameters update

### PR DESCRIPTION
**Pull request by patricktnast**
### Description

- *Category*: 
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-XYZ

### Changes and notes
 
The backups branch was not updating the job params object in place, which we need to do to save the correct metadata information.

### CODE REVIEW

**Review by stevebachmeier — APPROVED** [review]

**Review by albrja — APPROVED** [review]

File: src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py [hunk 1]
```diff
@@ -131,7 +138,7 @@ def work_horse(job_parameters: dict) -> Tuple[pd.DataFrame, Dict[str, pd.DataFra
         logger.info(f"Exiting job: {job_parameters}")
 
 
-def setup_sim(job_parameters: JobParameters) -> ParallelSimulationContext:
+def add_manager_configurations(job_parameters: JobParameters) -> JobParameters:
```
Comment (rmudambi) [inline src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py:141]: What does this have to do with managers? Isn't it about branch configurations?

**Review by rmudambi — APPROVED** [review]

**Comment by patricktnast** [comment]
tested in:
/ihme/homes/pnast/scratch/vivarium/vivarium_gates_nutrition_optimization_child/backup_testing/pakistan/2024_09_10_13_14_14
did:
`psimulate run model_specifications/nutrition_optimization_child.yaml model_specifications/branches/scenarios.yaml -o /ihme/homes/pnast/scratch/vivarium/vivarium_gates_nutrition_optimization_child/backup_testing -P proj_simscience --backup-freq 0.5`
with one draw and seed to completion
then edited the keyspace to include 2 draws and 2 seeds so a `restart` would pick up additional jobs and ran
`psimulate restart /ihme/homes/pnast/scratch/vivarium/vivarium_gates_nutrition_optimization_child/backup_testing/pakistan/2024_09_10_13_14_14/ --backup-freq 0.1 --project proj_simscience`
then killed these after a set of backups were created
then ran:
` psimulate restart /ihme/homes/pnast/scratch/vivarium/vivarium_gates_nutrition_optimization_child/backup_testing/pakistan/2024_09_10_13_14_14/ --backup-freq none --project proj_simscience`

`8635873.3.log` shows this job was pulled from the backup, and all jobs completed successfully without e.g. a keyerror.

File: src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py (hunk 1 above)
Comment (patricktnast) [reply src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py:141]: the thing it is adding to the `branch_configuration` is the configuration for the artifact and randomness managers. It seems like putting them in "branch_configuration" in the first place is kind of a misleading choice. Steve and I felt like this name conveys that intent better
